### PR TITLE
Backpack.tf (with huge changes and improvements)

### PR DESCRIPTION
**Pull request by Ali-Joon**
So I had worked on this long time ago but after some time I was busy with my exams and left this off and the pull request has been rejected then I completely forgot about it. Basically, this Presence is for the website backpack.tf, the biggest trading and price listing website for the game tf2.
It's been months since I left this presence development. until a few days ago which I went back. I've been working on it again and now this time, It works.
I'll send some proof images too. Hope everything is enough and good to go.
pre1
pre2

### CODE REVIEW

File: websites/B/backpack.tf/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  "$schema": "https://schemas.premid.app/metadata/1.4",
+  "service": "Backpack.tf",
+  "author": {
+    "name": "a＠a",
+    "id": "405028899798384640"
+  },
+  "category": "other",
+  "logo": "https://i.imgur.com/YtdujIW.png",
+  "thumbnail": "https://i.imgur.com/uQCICM4.png",
+  "version": "1.0.1",
```
Comment (EncryptedDev) [inline websites/B/backpack.tf/dist/metadata.json:11]: Why are you starting on 1.0.1?
```suggestion
  "version": "1.0.0",
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
You need to fix all those CodeFactor notices too!

File: websites/B/backpack.tf/presence.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
+const presence = new Presence({
+  clientId: "850295838361649153"
+}),
+browsingStamp = Math.floor(Date.now() / 1000);
+let title: any,
+    sec: any;
```
Comment (Slowlife01) [inline websites/B/backpack.tf/presence.ts:6]: No any

File: websites/B/backpack.tf/presence.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "logo"
+  };
+  presenceData.startTimestamp = browsingStamp;
+  
+  if (document.location.hostname === "backpack.tf") {
```
Comment (Slowlife01) [inline websites/B/backpack.tf/presence.ts:13]: You don't need this

File: websites/B/backpack.tf/presence.ts [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    else if (document.location.pathname.includes("/top/donators")) presenceData.details = "Viewing Top Donators";
+    else if (document.location.pathname.includes("/top/generous")) presenceData.details = "Viewing Top Gifters";
+    else if (document.location.pathname.includes("/top/contributors")) presenceData.details = "Viewing Top Contributors";
+    else if (document.location.pathname.includes("/developer")) presenceData.details = "Viewing Develoapers page";
+    else if (document.location.pathname.includes("/help")) presenceData.details = "Viewing help center";
+    else if (document.location.pathname.includes("/rules")) presenceData.details = "Reading rules";
+    else if (document.location.pathname.includes("/servers")) presenceData.details = "Viewing servers with backpack.tf plugin";
+    else if (document.location.pathname.includes("/top/accurate")) presenceData.details = "Viewing The Most Accurate Users";
+}
+  if (presenceData.details === null) {
```
Comment (Slowlife01) [inline websites/B/backpack.tf/presence.ts:130]:
```suggestion
  if (!presenceData.details) {
```

**Review by Slowlife01 — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**